docs/proposals/params.md

```diff
@@ -0,0 +1,205 @@
+# Parameters
+
+A parameter is a key-value pair that describes configuration information or information about
+another system entity.
+
+Modeling configuration parameters as an API resource has utility that cuts across several different
+use-cases.  Both system components in OpenShift/kubernetes and software running in pods can benefit
+from a central configuration store:
+
+1.  Cluster operators could store configuration data for system components, enabling components to
+    self-bootstrap with just a master URL
+2.  Users could model external shared services and consume information about them in pods without
+    having to redefine parameters in each container
+3.  Users could model supplemental information about services as parameters and consume them in the
+    pods that consumed those services
+4.  Users could store configuration info for their components as parameters and consume them in the
+    appropriate pods
+
+## Use-Case: Configuring system components
+
+Upstream issue [kubernetes/1627](https://github.com/GoogleCloudPlatform/kubernetes/issues/1627)
+discusses storing configuration for system components in a resource with a `map[string]string`
+structure.  If a component is configured with a namespace/name, it should be able to pull its
+configuration from the API server.  A `Parameters` resource would solve this use case; a system
+component would only need the URL for the API server and the coordinates for its configuration
+in order to discover its configuration.  It could also watch the `Parameters` resource and receive
+events when its configuration was changed.
+
+An alternative to a parameters resource that has been discussed for configuring system components
+upstream is to store configuration data in etcd directly.  Using `Parameters` to sure
+configuration about services has several advantages to storing them directly in etcd or using CLI
+arguments exclusively:
+
+1.  Eliminates the need to query etcd directly, resulting in lower complexity; consumers only need
```

Comment:
Consumers have to use a new api (ours) instead of etcd is a huge argument against this.  If this directly mapped to an underlying value string, it's less worrisome.  Etcd lets users support hundreds of bits of config.  Are we going to do that?  How complex is our scenario before it would be better solved by etcd?

Seek to actively constrain your use case so it cannot be seen as offering anything other than a very simple passthrough / integrated tool that maps cleanly to underlying uses.